docs/release-checklist.rst

```diff
@@ -0,0 +1,155 @@
+
+
+Create Release Candidate
+````````````````````````
+
+Before "officially" tagging any release, we will make a
+release-candidate available. So there will be at least 1.15.0rc0 (for
+example). If there are any problems, an rc1 or rc2 etc may also be
+released. Anyone can sign these releases (ideally they'd be signed
+"officially" as well, but it's better to get them out than to wait for
+that).
+
+Typically expert users will be the ones testing release candidates and
+they will need to evaluate which contributers' signatures they trust.
+
+- (all steps above are completed)
+- sign the release
+  - git tag -s -u 0xE34E62D06D0E69CFCA4179FFBDE0D31D68666A7A -m "release Tahoe-LAFS-1.15.0rc0" tahoe-lafs-1.15.0.rc0
+  - (replace the key-id above with your own)
+- build all code locally
+  - these should all pass:
+    - tox -e py27,codechecks,docs,integration
+  - these can fail (ideally they should not of course):
+    - tox -e deprecations,upcoming-deprecations
+- build tarballs
+  - tox -e tarballs
+  - confirm it at least exists:
+  - ls dist/ | grep 1.15.0rc0
+- inspect and test the tarballs
+  - install each in a fresh virtualenv
+  - run basic tests
+- when satisfied, sign the tarballs:
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0-py2-none-any.whl
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.tar.bz2
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.tar.gz
+  - gpg --pinentry=loopback --armor --sign dist/tahoe_lafs-1.15.0rc0.zip
+
+
+Privileged Contributor
+-----------------------
+
+Steps in this portion require special access to keys or
+infrastructure. For example, **access to tahoe-lafs.org** to upload
+binaries or edit HTML.
+
+
+Hack Tahoe-LAFS
+```````````````
+
+Did anyone contribute a hack since the last release? If so, then
+https://tahoe-lafs.org/hacktahoelafs/ needs to be updated.
+
+
+Upload Artifacts
+````````````````
+
+Any release-candidate or actual release plus signature (.asc file)
+need to be uploaded to https://tahoe-lafs.org in ~source/downloads
+
+- secure-copy all release artifacts to the download area on the
+  tahoe-lafs.org host machine. `~source/downloads` on there maps to
+  https://tahoe-lafs.org/downloads/ on the Web.
+- scp dist/*1.15.0* meejah@tahoe-lafs.org:/home/source/downloads
+- the following developers have access to do this:
+  - exarkun
+  - meejah
+  - warner
+
+For the actual release, the tarball and signature files need to be
+uploaded to PyPI as well.
+
+- how to do this?
+- (original guide says only "twine upload dist/*")
+- the following developers have access to do this:
+  - exarkun
+  - warner
+
+
+Upload Dependencies
+```````````````````
+
+The original guide says, "upload wheels to
+https://tahoe-lafs.org/deps/" which seems to be all the wheels of all
+the dependencies. There are no instructions on how to collect these or
+where to put them on the tahoe-lafs.org machines.
+
+Is this step still useful?
```

Comment:
I support dropping this one.

Reply:
Should be delete the stuff that's on tahoe-lafs.org now as well? Anyway I filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3494#ticket

Reply:
I guess there's a chance someone is operating an older release according to older installation instructions and could get mad if the tahoe-lafs.org-hosted artifacts vanished...

If it is real, that is an unfortunate situation.  Perhaps we can determine if it is real by inspecting the HTTP access logs on tahoe-lafs.org - if there are any.

Hm.  There are access logs and there are some hits on `/deps/` from something claiming to be pip.  So maybe no clear case to be made to delete that stuff right away.  Probably should update the install docs that refer to that URL and get that into a release first, at minimum.